model/ftn/w3ounfmetamd.ftn

```diff
@@ -0,0 +1,2893 @@
+
+      CHARACTER(LEN=*), INTENT(INOUT) :: INSTR
+      INTEGER, INTENT(IN) :: IPART
+!/
+!/ ------------------------------------------------------------------- /
+!/ Local parameters
+!/
+      INTEGER :: I, J
+
+      I = INDEX(INSTR, "<IPART>")
+      J = I + LEN_TRIM("<IPART>")
+      IF(I .GT. 0) THEN
+        WRITE(INSTR,'(A,I1,A)') INSTR(1:I-1), IPART, INSTR(J:LEN(INSTR))
+      ENDIF
+
+      END SUBROUTINE PARTNO_STRING_SUB
+
+!/ ------------------------------------------------------------------- /
+      SUBROUTINE WRITE_META(NCID, VARID, META, ERR)
+!/
+!/                  +-----------------------------------+
+!/                  | WAVEWATCH III           NOAA/NCEP |
+!/                  |           C. Bunney               |
+!/                  |                                   |
+!/                  |                        FORTRAN 90 |
+!/                  | Last update :         02-Nov-2020 |
+!/                  +-----------------------------------+
+!/
+!/    09-Nov-2020 : Creation                            ( version 7.XX )
+!/
+!
+!  1. Purpose :
+!
+!     Writes the meta-data entries for a variable.
+!
+!  2. Parameters :
+!
+!     Parameter list
+!     ----------------------------------------------------------------
+!       NCID    Int.  I/O  NetCDF file ID
+!       VARID   Int.  I/O  NetCDF variable ID
+!       META    Int.  I    Meta data type
+!       ERR     Int.  O    Error value
+!     ----------------------------------------------------------------
+!
+!/ ------------------------------------------------------------------- /
+      IMPLICIT NONE
+
+      INTEGER, INTENT(IN) :: NCID, VARID
+      TYPE(META_T), INTENT(IN) :: META
+      INTEGER, INTENT(OUT) :: ERR
+!/
+!/ ------------------------------------------------------------------- /
+!/ Local parameters
+!/
+      INTEGER :: IVAL, VAL
+      REAL :: RVAL
+!/
+      ERR = NF90_PUT_ATT(NCID, VARID, 'long_name', META%VARNL)
+      IF(ERR /= NF90_NOERR) RETURN
+      
+      ERR = NF90_PUT_ATT(NCID, VARID, 'standard_name', META%VARNS)
+      IF(ERR /= NF90_NOERR) RETURN
+      
+      IF(META%VARNG .NE. '' .AND. META%VARNG .NE. UNSETC) THEN
+        ERR = NF90_PUT_ATT(NCID, VARID, 'globwave_name', META%VARNG)
+        IF(ERR /= NF90_NOERR) RETURN
+      ENDIF
+      
+      ERR = NF90_PUT_ATT(NCID, VARID, 'units', META%UNITS)
+      IF(ERR /= NF90_NOERR) RETURN
+
+      ! Fill value dependent on variable type
+      IF(NCVARTYPE .EQ. 2) THEN 
+        ERR = NF90_PUT_ATT(NCID, VARID, '_FillValue', NF90_FILL_SHORT)
+      ELSE 
+        ERR = NF90_PUT_ATT(NCID, VARID, '_FillValue', NF90_FILL_FLOAT)
+      END IF
+      IF(ERR /= NF90_NOERR) RETURN
+
+      ERR = NF90_PUT_ATT(NCID, VARID, 'scale_factor', META%FSC)
+      IF(ERR /= NF90_NOERR) RETURN
+      
+      ERR = NF90_PUT_ATT(NCID, VARID, 'add_offset', 0.)
+      IF(ERR /= NF90_NOERR) RETURN
+      
+      ! Note: Valid min/max are scaled by scale_factor 
+      !       and converted to integers.
```

Comment:
That's how it was before, but you're right - I should only be scaling to integer for NF90_SHORT data types.
I will update so if NCVARTYPE is 2, it is scaled and converted to integer, otherwise it is written straight out as a float.

Reply:
great

Reply:
OK - done for commit 45f56ef4